hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/fs/shell/TestBulkDeleteCommand.java

```diff
@@ -27,103 +27,134 @@
-                describedAs("Children arguments must match").isEqualTo(arg2);
-    }
+  @Test
+  public void testArguments() throws IOException, URISyntaxException {
+    BulkDeleteCommand bulkDeleteCommand = new BulkDeleteCommand(conf);
+    LinkedList<String> arguments = new LinkedList<>();
+    String arg1 = "file:///file/name/1";
+    String arg2 = "file:///file/name/1/2";
+    arguments.add(arg1);
+    arguments.add(arg2);
+    LinkedList<PathData> pathData = bulkDeleteCommand.expandArguments(arguments);
+    Assertions.assertThat(pathData.size()).
+            describedAs("Only one root path must be present").isEqualTo(1);
+    Assertions.assertThat(pathData.get(0).path.toUri().getPath()).
+            describedAs("Base path of the command should match").isEqualTo(new URI(arg1).getPath());
+    Assertions.assertThat(bulkDeleteCommand.childArgs.size()).
+            describedAs("Only one other argument was passed to the command").
+            isEqualTo(1);
+    Assertions.assertThat(bulkDeleteCommand.childArgs.get(0)).
+            describedAs("Children arguments must match").isEqualTo(arg2);
+  }
 
-    @Test
-    public void testLocalFileDeletion() throws IOException {
-        String deletionDir = "toDelete";
-        String baseFileName = "file_";
-        Path baseDir = new Path(testRootDir, deletionDir);
-        List<String> listOfPaths = new ArrayList<>();
-        for(int i = 0; i < 100; i++) {
-            Path p = new Path(baseDir, baseFileName + i);
-            lfs.create(p);
-            listOfPaths.add(p.toUri().toString());
-        }
-        List<String> finalCommandList = new ArrayList<>();
-        finalCommandList.add("-bulkDelete");
-        finalCommandList.add(baseDir.toUri().toString());
-        finalCommandList.addAll(listOfPaths);
-        shell.run(finalCommandList.toArray(new String[0]));
-        Assertions.assertThat(lfs.listFiles(baseDir, false).hasNext())
-                .as("All the files should have been deleted").isEqualTo(false);
+  @Test
+  public void testWrongArguments() throws IOException, URISyntaxException {
+    BulkDeleteCommand bulkDeleteCommand = new BulkDeleteCommand(conf);
+    LinkedList<String> arguments = new LinkedList<>();
+    String arg1 = "file:///file/name/1";
+    arguments.add(arg1);
+    Assertions.assertThatThrownBy(() -> bulkDeleteCommand.expandArguments(arguments)).
+            describedAs("No children to be deleted specified in the command.").
+            isInstanceOf(IOException.class);
+  }
 
+  @Test
+  public void testLocalFileDeletion() throws IOException {
+    String deletionDir = "toDelete";
+    String baseFileName = "file_";
+    Path baseDir = new Path(testRootDir, deletionDir);
+    List<String> listOfPaths = new ArrayList<>();
+    for (int i = 0; i < 100; i++) {
+      Path p = new Path(baseDir, baseFileName + i);
+      ContractTestUtils.touch(lfs, p);
     }
+    RemoteIterator<LocatedFileStatus> remoteIterator = lfs.listFiles(baseDir, false);
```

Comment:
Add assertion that files are created successfully.

Reply:
Done

Reply:
you have to push the latest commit I guess